### ISSUE TITLE
ServiceBinding configuration for Kafka Streams

### ISSUE THREAD

**Issue by wtrocki** [quarkusio/quarkus#17006]
## Describe the bug

This is more of a question to @cescoffier rather than bug.
Kafka streams require a different configurations for the producer (authentication details need to be duplicated).
Do we want Kafka Service bindings to work with Kafka Connect/Kafka Streams etc. 

My take is that we actually document what type of properties are provided by binding and they can be reused. 
Then this issue would be duplicate of #16737 

```
# Used by Kafka Connnect AdminClient
sasl.mechanism=PLAIN
security.protocol=SASL_SSL
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="srvc-acct-123" password="";

# Used by the created Kafka Connect Producer
producer.security.protocol=SASL_SSL
producer.sasl.mechanism=PLAIN
producer.sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="srvc-acct-123" password="";

**Comment by cescoffier** [comment]
I don't believe we support Connect at all. There is no extension for that, and looking at your snipped, the second part uses a different structure ("producer" prefix). The first part of the snippet should work if this admin client is created from the default Kafka broker configuration.

**Comment by wtrocki** [comment]
So from user point of view if they want to work with external sources we probably need some intermediary property that can be reused by users for any upstream lib out there. Is that something that would be good for quarkus or outside scope?

**Comment by cescoffier** [comment]
What do you mean by _ intermediary_?

The integration at the moment exposes a Map<String, Object> that you can retrieve and use to configure all Kafka-related objects (consumer, admin, producer)... The map does not need to be complete, it's often not (and up to the user to complete the map with the relevant serdes and so on).
